### PR TITLE
docs: add interactive notebooks for Linear Regression and Logistic Regression

### PR DESCRIPTION
**Pull request by Larissa574**
Ajout des notebooks de documentation interactifs pour les modules de régression 
linéaire et de régression logistique, en suivant le template établi par le groupe 
KNN. Le dossier notebooks/regression/ n'existait pas encore dans le projet, il a 
donc été créé pour l'occasion. Chaque notebook couvre les concepts clés avec 
formules mathématiques, le pseudo-algorithme, un benchmark ifri-mini-ml-lib vs 
scikit-learn, une démo interactive avec ipywidgets, ainsi que les applications 
réelles, limitations et références. Les datasets utilisés sont California Housing 
pour la régression linéaire et Breast Cancer Wisconsin pour la régression logistique.

### CODE REVIEW

**Review by rosasbehoundja — CHANGES_REQUESTED** [review]
Effort appréciable !

Dans les deux notebooks, veuillez à ajouter des précisions concernant la descente de gradient et certaines images pour servir d'illustrations.

**Comment by kalil-236** [comment]
J'ai créer un nouveau fichier naive_bayes.py dans le dossier classification , toute l'implémentation ya été effectuée. Suite à celà j'ai créé une suite de tests unitaires avec Pytest couvrant l'ajustement du modèle (fit), la cohérence des prédictions et la gestion des données via une fixture dédiée. Tous les tests unitaires (4/4) sont validés localement.

**Comment by Larissa574** [comment]
- Ajout de l'intuition de la descente de gradient et d'une illustration de la fonction de perte en forme de bol dans le notebook de Régression Linéaire
- Ajout de l'illustration de la courbe sigmoïde et des précisions sur la descente de gradient avec momentum dans le notebook de Régression Logistique

**Comment by Sylvere-Kinox** [comment]
J'ai implémenté les modèles `DecisionTreeRegressor` et `RandomForestRegressor` avec échantillonnage bootstrap et moyenne des prédictions. J'ai ajouté des tests unitaires avec pytest pour valider l'initialisation, l'entraînement, l'échantillonnage bootstrap et les résultats de prédiction du modèle.

**Comment by femi-cloud** [comment]
J'ai ajouté les notebooks de documentation interactifs pour le Random Forest Regressor et le Naive Bayes, en suivant le template établi par le groupe KNN. Chaque notebook couvre les concepts clés avec formules mathématiques, les pseudo-algorithmes, un benchmark ifri-mini-ml-lib vs scikit-learn, une démo interactive avec ipywidgets, ainsi que les applications
réelles, limitations et références. Les datasets utilisés sont Iris dataset pour Naive Bayes et des données synthétiques (dataset artificiel) pour le Random Forest.

**Comment by femi-cloud** [comment]
J'ai ajouté l'implémentation et la démo de la classification au notebook de Random Forest.

**Comment by Ulk-bm2k25** [comment]
J'ai ajouté un nouveau fichier random_forest.py dans le dossier nommé classification. L'implémentation repose directement sur la classe  DecisionTree existante, importée et instanciée pour chaque arbre de la forêt. L'algorithme utilise "sqrt" (  √(n_features)  ) par défaut comme stratégie de sélection des features pour chaque arbre. Les prédictions finales sont déterminées par vote majoritaire via predict, et les probabilités par classe sont disponibles via predict_proba. Le modèle supporte la classification binaire et multi-classes. Suite à cela, j'ai créé une suite de tests unitaires avec Pytest couvrant l'ajustement du modèle (fit), la cohérence des prédictions, la validité des probabilités retournées et la gestion des erreurs, avec des données d'entraînement organisées via des fixtures dédiées. Tous les tests unitaires (7/7) sont validés localement.

**Comment by medard-apl** [comment]
J'ai ajouté plus haut les NoteBooks interactifs pour Decision Tree Regressor, Polynomial Regression et SVR.
Ces derniers suivent le template de base fourni pour KNN.
Il y a donc les concepts clés, les caractères et critères propres à chaque algorithme, un pseudo algorithme, l'implémentation (SKlearn et Ifri_mini_ml_lib), une démo puis les applications et limites.

File: ifri_mini_ml_lib/regression/decision_tree_regressor.py [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+        return self._mse(y) - (n_left/n  * self._mse(y[left_mask]) +
+                                n_right/n * self._mse(y[right_mask]))
+
+    def _leaf_value(self, y):
+        """Retourne la moyenne des y — valeur de la feuille."""
+        return np.mean(y)
+
+    def predict(self, X):
+        if self.tree is None:
+            raise ValueError("Le modèle doit être entraîné avant de faire des prédictions.")
```
Comment (rosasbehoundja) [inline ifri_mini_ml_lib/regression/decision_tree_regressor.py]: Warning en anglais :)

**Review by rosasbehoundja — CHANGES_REQUESTED** [review]
Excellent travail !!!

1. Pour la documentation faire une doc globale pour le SVM et expliquer juste comment ça se passe pour les cas de regression ou classification. Pareil pour les Decisions Trees.

**Comment by medard-apl** [comment]
Bien reçu


Le sam. 16 mai 2026 à 21:48, Rosas Behoundja ***@***.***> a
écrit :

**Comment by medard-apl** [comment]
J'ai amélioré la documentation pour Arbre de décision. Celle ci aborde maintenant et la classification et la régression et a été envoyée dans la section Notebook/Classification.
J'ai ensuite supprimé l'ancien fichier decision_tree_regressor.ipynb initialement prévu uniquement pour la régression.

**Comment by femi-cloud** [comment]
J'ai déplacé le notebook de random forest de notebooks/regression/ vers notebooks/classification.

**Comment by medard-apl** [comment]
J'ai ajouté des schémas et graphes illustratifs dans les notebooks : SVR, Arbre de décision et Régression Polynomiale.
Cela a impliqué la création de dossiers et le stockage des images dans notebook/assets/imgs.

**Comment by Larissa574** [comment]
Déplacement du  notebooks dans le dossier approprié et ajout de la comparaison SVR/SVM dans le notebook SVR.
